instrumentation/github.com/aws/aws-lambda-go/otellambda/lambda.go

```diff
@@ -56,6 +57,23 @@ func newInstrumentor(opts ...Option) instrumentor {
 		resAttrs: []attribute.KeyValue{}}
 }
 
+func xrayEnvToLinks() []trace.Link {
```

Comment:
Not sure if this is idiomatic to return a list of one item or an empty list.

In some languages I'd definitely do it this way, in others I'd return null and others I'd return a `Maybe Link`, with `WithLinks` accepting those.

Returning `nil` or `*trace.Link` felt off (especially since I didn't see a way to not have to do one `start` in the first branch of `if link != nil` and another in the else branch), but maybe that is the way to go to avoid an unnecessary creation of an empty list?